### PR TITLE
api(ticdc): Not show the error info during query the changefeed when state is stopped, finished and removed

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/11642

### What is changed and how it works?
Make the previous error message not return when changefeed state is stopped / finished / removed, to avoid confused.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/api/v2/changefeed.go [hunk 1]
```diff
@@ -284,7 +284,7 @@ func (h *OpenAPIV2) listChangeFeeds(c *gin.Context) {
 
 		// if the state is normal, we shall not return the error info
 		// because changefeed will is retrying. errors will confuse the users
-		if commonInfo.FeedState == model.StateNormal {
+		if commonInfo.FeedState == model.StateNormal || commonInfo.FeedState == model.StateStopped || commonInfo.FeedState == model.StateFinished || commonInfo.FeedState == model.StateRemoved {
```
Comment (sdojjy) [inline cdc/api/v2/changefeed.go:287]: too complex here, exact it to a function, then use it in line 1039

**Review by asddongmen — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Review by benmeadowcroft — APPROVED** [review]

**Pull request merged**